[17.0][TMP-FIX] sale_purchase_order_line_report: Temporal fix for sale orde…

DantePereyra opened the pull request:
…r line report

Before:
- The report only considered sales lines in state `sale`. Now:
- All lines in `sale` are included.
- Lines in `draft` or `sent` are also included, but only if `qty_delivered > 0`.

This is a TMP-FIX to cover the issue detected in `pos_sale`, where some quotations stay in draft/sent state but already have delivered quantities.

dalonsod reviewed (approved):
Tested :+1: 

Please add in `ROADMAP` this temporary fix, that should be removed when original Odoo bug were fixed.

dalonsod commented:
Please add pending commit with `README` & `index.html` update, and I'll merge it

dalonsod commented:
Remember, UPD commit for main README and index files should be a standalone one, not merged with functional commits

DantePereyra committed:
[TMP-FIX] sale_purchase_order_line_report: Temporal fix for sale order line report
Before:
- The report only considered sales lines in state .
Now:
- All lines in  are included.
- Lines in  or  are also included, but only if .

This is a TMP-FIX to cover the issue detected in ,
where some quotations stay in draft/sent state but already have delivered quantities.

DantePereyra committed:
[UPD] sale_purchase_order_line_report: README.rst and index.html
